### PR TITLE
GH-164 Removed warning for new purchase options and added warning on publish

### PR DESCRIPTION
**Pull request by milindmore22**
related to #164

### CODE REVIEW

File: assets/src/js/pages/revenue-generator-paywall.js [hunk 1]
```diff
@@ -1097,8 +1097,17 @@ import { __, sprintf } from '@wordpress/i18n';
...
 					);
 
+					let entityId;
+					if ( 'individual' === optionType ) {
+						entityId = optionItem.attr( 'data-paywall-id' );
+					} else if ( 'timepass' === optionType ) {
+						entityId = optionItem.attr( 'data-tlp-id' );
+					} else if ( 'subscription' === optionType ) {
+						entityId = optionItem.attr( 'data-sub-id' );
+					}
```
Comment (tomasstark) [inline assets/src/js/pages/revenue-generator-paywall.js:1107]: Maybe we can consider converting this `if / else if` block to `switch` statement for better readability?

File: assets/src/js/pages/revenue-generator-paywall.js [hunk 2]
```diff
@@ -1300,8 +1309,17 @@ import { __, sprintf } from '@wordpress/i18n';
...
 							}
 
+							let entityId;
+							if ( 'individual' === optionType ) {
+								entityId = optionItem.attr( 'data-paywall-id' );
+							} else if ( 'timepass' === optionType ) {
+								entityId = optionItem.attr( 'data-tlp-id' );
+							} else if ( 'subscription' === optionType ) {
+								entityId = optionItem.attr( 'data-sub-id' );
+							}
```
Comment (tomasstark) [inline assets/src/js/pages/revenue-generator-paywall.js:1319]: Can this be converted to `switch` statement so the code is more readable?

File: assets/src/js/pages/revenue-generator-paywall.js [hunk 3]
```diff
@@ -2126,18 +2144,74 @@ import { __, sprintf } from '@wordpress/i18n';
...
+							const purchaseType = optionItem.attr(
+								'data-purchase-type'
+							);
+							if ( 'subscription' === purchaseType ) {
+								entityId = optionItem.attr( 'data-sub-id' );
+							} else if ( 'timepass' === purchaseType ) {
+								entityId = optionItem.attr( 'data-tlp-id' );
+							} else if ( 'individual' === purchaseType ) {
+								entityId = optionItem.attr(
+									'data-paywall-id-id'
```
Comment (tomasstark) [inline assets/src/js/pages/revenue-generator-paywall.js:2165]: Is `data-paywall-id-id` intentional here (notice `id` is there twice) or a typo?

File: assets/src/js/pages/revenue-generator-paywall.js [hunk 4]
```diff
@@ -2126,18 +2144,74 @@ import { __, sprintf } from '@wordpress/i18n';
...
+
+						$.each( purchaseOptionItems, function(
+							key,
+							purchaseOptionItem
+						) {
+							const optionItem = $( purchaseOptionItem );
+							const purchaseType = optionItem.attr(
+								'data-purchase-type'
+							);
+							if ( 'subscription' === purchaseType ) {
```
Comment (tomasstark) [inline assets/src/js/pages/revenue-generator-paywall.js:2159]: Maybe we can consider converting this `if / else if` block to `switch` statement for better readability?

**Review by tomasstark — APPROVED** [review]
LGTM 🚀 with some comments.

File: assets/src/js/pages/revenue-generator-paywall.js (hunk 3 above)
Comment (milindmore22) [reply assets/src/js/pages/revenue-generator-paywall.js]: Good catch it was a typo

**Comment by milindmore22** [comment]
@tomasstark Addressed PR feedback please review again

**Review by tomasstark — APPROVED** [review]
Perfect! 👍

**Pull request merged**